cardano-config/src/Cardano/Config/Protocol.hs

```diff
@@ -74,49 +82,50 @@ type TraceConstraints blk =
 mockSecurityParam :: SecurityParam
 mockSecurityParam = SecurityParam 5
 
-data SomeProtocol where
-  SomeProtocol :: (RunNode blk, TraceConstraints blk)
-               => Consensus.Protocol blk -> SomeProtocol
+-- | Helper for creating a 'SomeProtocol' for a mock protocol that needs the
+-- 'CoreNodeId' and NumCoreNodes'. If one of them is missing from the
+-- 'CardanoConfiguration', a 'MissingNodeInfo' exception is thrown.
+mockSomeProtocol
+  :: (RunNode blk, TraceConstraints blk)
+  => CardanoConfiguration
+  -> (CoreNodeId -> NumCoreNodes -> Consensus.Protocol blk)
+  -> IO SomeProtocol
+mockSomeProtocol cc mkConsensusProtocol =  do
+    (cid, numCoreNodes) <- either throwIO return $ extractNodeInfo cc
+    let p = mkConsensusProtocol cid numCoreNodes
+    case Consensus.runProtocol p of
+      Dict -> return $ SomeProtocol p
 
-fromProtocol :: CardanoConfiguration -> Protocol -> IO SomeProtocol
 
-fromProtocol _ ByronLegacy =
-  error "Byron Legacy protocol is not implemented."
 
-fromProtocol _ BFT =
-  case Consensus.runProtocol p of
-    Dict -> return $ SomeProtocol p
+data SomeProtocol where
+  SomeProtocol :: (RunNode blk, TraceConstraints blk)
+               => Consensus.Protocol blk -> SomeProtocol
 
-  where
-    p = Consensus.ProtocolMockBFT mockSecurityParam
+fromProtocol :: CardanoConfiguration -> IO SomeProtocol
+fromProtocol cc = case ccProtocol cc of
+  ByronLegacy ->
+    error "Byron Legacy protocol is not implemented."
 
-fromProtocol _ Praos =
-  case Consensus.runProtocol p of
-    Dict -> return $ SomeProtocol p
+  BFT         -> mockSomeProtocol cc $ \cid numCoreNodes ->
+    Consensus.ProtocolMockBFT numCoreNodes cid mockSecurityParam
 
-  where
-    p = Consensus.ProtocolMockPraos PraosParams {
+  Praos       -> mockSomeProtocol cc $ \cid numCoreNodes ->
+    Consensus.ProtocolMockPraos numCoreNodes cid PraosParams {
         praosSecurityParam = mockSecurityParam
       , praosSlotsPerEpoch = 3
       , praosLeaderF       = 0.5
       , praosLifetimeKES   = 1000000
       }
 
-
-fromProtocol _ MockPBFT =
-  case Consensus.runProtocol p of
-    Dict -> return $ SomeProtocol p
-
-  where
-    p = Consensus.ProtocolMockPBFT PBftParams {
+  MockPBFT    -> mockSomeProtocol cc $ \cid numCoreNodes@(NumCoreNodes numNodes) ->
+    Consensus.ProtocolMockPBFT numCoreNodes cid PBftParams {
         pbftSecurityParam      = mockSecurityParam
-      , pbftNumNodes           = numNodes
+      , pbftNumNodes           = fromIntegral numNodes
       , pbftSignatureThreshold = (1.0 / fromIntegral numNodes) + 0.1
       }
-    numNodes = 3
```

Comment:
The number of nodes used to be hard-coded to 3 for MockPBFT. Now it will use the value from the config.